src/interface/emacs/khoj.el

```diff
@@ -416,10 +416,47 @@ Auto invokes setup steps on calling main entrypoint."
          (files-to-index (or file-paths
                              (append (mapcan (lambda (dir) (directory-files-recursively dir "\\.\\(org\\|md\\|markdown\\|pdf\\|txt\\|rst\\|xml\\|htm\\|html\\)$")) content-directories) content-files)))
          (type-query (if (or (equal content-type "all") (not content-type)) "" (format "t=%s" content-type)))
+         (delete-files (khoj--get-delete-file-list khoj--indexed-files files-to-index))
          (inhibit-message t)
-         (message-log-max nil))
-    (let ((url-request-method "POST")
-          (url-request-data (khoj--render-files-as-request-body files-to-index khoj--indexed-files boundary))
+         (message-log-max nil)
+         (batch-size 30))
+	(dolist (files (khoj--split-file-list files-to-index batch-size))
+      (khoj--send-index-update-request (khoj--render-update-files-as-request-body files boundary) boundary content-type type-query force))
+    (when delete-files
+        (khoj--send-index-update-request (khoj--render-delete-files-as-request-body delete-files boundary) boundary content-type type-query force))
+    (setq khoj--indexed-files files-to-index)))
+
+(defun khoj--get-delete-file-list (indexed-files upload-files)
+  "Get delete file list. when `INDEXED-FILES' no longer in `UPLOAD-FILES'.
+delete them. return delete-file-list."
+  (let (delete-files '())
+    (dolist (indexed-file indexed-files)
+      (when (not (member indexed-file upload-files))
+        (push indexed-file delete-files)))
+    delete-files))
+
+(defun khoj--split-file-list (file-list size)
+  "Split `FILE-LIST' into subgroups of `SIZE' files each."
+  (let ((subgroups '())
+        (current-group '()))
+    (dolist (file file-list)
+      (if (= (length current-group) size)
+          ;; If the current group has size files, start a new group
+          (progn
+            (push current-group subgroups)
+            (setq current-group '()))
+        (push file current-group)))
+    ;; Add the last group if it's not empty
+    (when current-group
+      (push (nreverse current-group) subgroups))
```

Comment:
Why is the last `current-group` reversed before the push into subgroups? The previous `current-group`'s (in line `446`) aren't being reversed before pushing into `subgroups`